### DISCUSSION TITLE
Ökofen Integration/deutsche Sonderzeichen

### DISCUSSION THREAD

**Discussion by MrPete2706** [dominikamann/oekofen-pellematic-compact#67] [Q&A]
Hallo Herr Amann,
vielen Dank für die Ökofen-Integration, ich habe sie gerade installiert und getestet, bin begeistert.
Ein kleines Problem habe ich bei der Darstellung der Statusmeldungen mit deutschen Sonderzeichen.
Habe ich etwas übersehen bzw. gibt es dazu eine Abhilfe? 

Beste Grüße aus Salzburg!

Peter
![2024-09-30 09_18_03-Overview – Home Assistant](https://github.com/user-attachments/assets/c83ddd0b-9740-4e4d-88f7-362cbeb273dc)

**Comment by dominikamann** [comment]
Hallo Peter,

vielen Dank! Ich kann den Fehler leider nicht nachvollziehen. Ist eventuell die Heizung auf einem alten Softwarestand? Was gibt die Rest-Schnittstelle direkt zurück?

Viele Grüße 

![image](https://github.com/user-attachments/assets/8870f588-49a9-43f2-bb77-355f3ea39f76)

**Comment by MrPete2706** [comment]
Hallo Dominik,
vielen Dank für die schnelle Antwort. Meine Anlage ist neu, wurde gerade erst in Betrieb genommen. Die Touch-Version ist 4.04.
Den Schnittstellen-Response werde ich testen und darüber hier berichten, nochmals vielen Dank für deine tolle Arbeit!
Peter

Habe nun die Schnittstelle direkt über den Browser angesprochen, die Sonderzeichen werden dort korrekt dargestellt, siehe Anhänge.
Gleichzeitig habe ich im Home Assistant diese Phantasiezeichen. Bin im Moment etwas ratlos.

![2024-09-30 16_20_38-Overview – Home Assistant](https://github.com/user-attachments/assets/9fcc8def-5985-494b-979a-90e971ff05ec)
![2024-09-30 16_21_25-Mozilla Firefox](https://github.com/user-attachments/assets/f7d91b6c-380d-4295-a1f5-34c940f74d60)
![2024-09-30 16_22_07-Mozilla Firefox](https://github.com/user-attachments/assets/803b5fdc-32b4-47a7-8df1-9716d3f97aff)

Noch eine Information: in der Sensorliste der Integration werden die Sonderzeichen bereits falsch dargestellt:
![2024-09-30 18_25_43-Settings – Home Assistant](https://github.com/user-attachments/assets/3c57d086-ee15-4787-8b6c-ee606f549ec3)

**Reply by dominikamann** [reply]
Sehr gerne 👍 Das ist sehr spannend da die Werte (Umlaute & Co.) bei mir korrekt übernommen sowie dargestellt werden. Ich bin jedoch auf Version V4.02b. Da es im Browser korrekt dargestellt wird ist vermutlich ein Zeichensatz Problem. Eventuell wurde die Schnittstelle seitens Ökofen verändert.

Aktuell decodieren wir mit "iso-8859-1". Spannend wäre zu wissen was der neue Zeichensatz ist dann kann ich mir eine Lösung einfallen lassen. Gibt der Browser etwas an Infos her?

Danke!

**Reply by MrPete2706** [reply]
Hallo Dominik,
im Browser konnte ich nichts finden, aber ich habe die Datei dann ins Notepad++ geladen, dort bekomme ich als Codierung UTF-8 angezeigt. Mein Verdacht war auch, dass hier ein Zeichensatz-Problem vorliegt, aber mir fehlt im Moment ein Ansatzpunkt zur Fehlerbehebung.
mbG
Peter

**Reply by dominikamann** [reply]
Ich schau mir das genauer an sobald ich wieder Zugriff auf meinen Rechner habe. Ich bin aktuell nicht zu Hause 🧳. Wir finden sicher eine Lösung ;) Ich melde mich 👍

**Reply by MrPete2706** [reply]
Hallo Dominik,
vielen Dank für deine Hilfe, habe dir soeben ein paar Biere spendiert!
Peter

**Reply by dominikamann** [reply]
Herzlichen Dank! Das wäre nicht nötig gewesen. Die Biere 🍻 werde ich genießen 💪👍

**Comment by dominikamann** [comment]
Hi Peter,

bitte probiere / teste Version https://github.com/dominikamann/oekofen-pellematic-compact/releases/tag/v3.3.6. Es ist nun möglich ein Charset (utf-8) anzugeben.

Danke.

**Reply by dominikamann** [reply]
![image](https://github.com/user-attachments/assets/53b03bff-5352-46e9-ae06-236735d76b6f)

**Comment by MrPete2706** [comment]
Hallo Dominik,
habe  die 3.3.6 soeben getestet, mit utf-8 funktioniert nun alles perfekt!
Danke für deine Hilfe, Kaltgetränke sind bereits unterwegs!
Peter
![2024-11-07 15_49_39-Window](https://github.com/user-attachments/assets/e86950f7-23f0-42c4-9ec5-2bf671a2fb24)

**Reply by dominikamann** [reply]
Das freut mich sehr das es nun geht 👍🎉. Ich konnte es selber nicht testen. Und Danke 👍
